### ISSUE TITLE
Quick input list dom nodes hang around after quick input dismissed

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode#163411]
Type: <b>Bug</b>

1. Open dev tools
1. Open quick input and start typing a file name
2. Close the quick input with escape
1. In the devtools, look at the quick input dom

**Bug**
The list elements for the previously showing quick input are still around

![Screen Shot 2022-10-12 at 1 12 30 AM](https://user-images.githubusercontent.com/12821956/195288816-d8feecd0-4469-40d4-a87e-2b4540b4fbdb.png)

While they are cleaned up once the quick input shows again, for performance it would be better to free them once the input closes and they are no longer needed

VS Code version: Code - Insiders 1.73.0-insider (Universal) (bd1b7a384f8a5ab9fb5a6a0e8fea370e589f43a3, 2022-10-11T09:27:32.881Z)
OS version: Darwin x64 21.6.0
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|5, 5, 5|
|Memory (System)|32.00GB (0.03GB free)|
|Process Argv|--crash-reporter-id 48781ca2-1705-4f64-9bab-325055aab55d|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by TylerLeonhardt** [comment]
I think this works the way that it does because it's possible to call `show` on a quick pick when a quick pick is already visible so the logic was put in the `show`. We can probably move it to a helper function and run it in both places.

**Comment by TylerLeonhardt** [comment]
@mjbvz didn't you fix this one in a PR? I can't seem to find it.

**Comment by mjbvz** [comment]
@TylerLeonhardt Nope. Not too familiar with the quick fix code so was waiting for you

**Comment by TylerLeonhardt** [comment]
reopening since I need to revert the fix: https://github.com/microsoft/vscode/pull/168489
